src/lib.rs

```diff
@@ -31,6 +31,24 @@ pub struct Header {
     pub value: Cow<'static, str>,
 }
 
+impl From<service_protocol::messages::Header> for Header {
```

Comment:
Please move these two implementations under the comment "other conversions" in service_protocol/messages.rs